stock_scrap/models/stock_scrap.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Odoo
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models, _
+from openerp.exceptions import UserError
+
+
+class StockScrap(models.Model):
+    _name = 'stock.scrap'
+    _order = 'id desc'
+
+    def _get_default_scrap_location_id(self):
+        return self.env['stock.location'].search(
+            [('scrap_location', '=', True)], limit=1).id
+
+    def _get_default_location_id(self):
+        return self.env.ref(
+            'stock.stock_location_stock', raise_if_not_found=False)
+
+    name = fields.Char(
+        'Reference',  default=lambda self: _('New'),
+        copy=False, readonly=True, required=True,
+        states={'done': [('readonly', True)]})
+    origin = fields.Char(string='Source Document')
+    product_id = fields.Many2one(
+        'product.product', 'Product',
+        required=True, states={'done': [('readonly', True)]})
+    product_uom_id = fields.Many2one(
+        'product.uom', 'Unit of Measure',
+        required=True, states={'done': [('readonly', True)]})
+    tracking = fields.Selection(
+        'Product Tracking', readonly=True, related="product_id.tracking")
+    lot_id = fields.Many2one(
+        'stock.production.lot', 'Lot',
+        states={'done': [('readonly', True)]},
+        domain="[('product_id', '=', product_id)]")
+    package_id = fields.Many2one(
+        'stock.quant.package', 'Package',
+        states={'done': [('readonly', True)]})
+    owner_id = fields.Many2one(
+        'res.partner', 'Owner', states={'done': [('readonly', True)]})
+    move_id = fields.Many2one('stock.move', 'Scrap Move', readonly=True)
+    picking_id = fields.Many2one(
+        'stock.picking', 'Picking', states={'done': [('readonly', True)]})
+    location_id = fields.Many2one(
+        'stock.location', 'Location', domain="[('usage', '=', 'internal')]",
+        required=True, states={'done': [('readonly', True)]},
+        default=_get_default_location_id)
+    scrap_location_id = fields.Many2one(
+        'stock.location', 'Scrap Location',
+        default=_get_default_scrap_location_id,
+        domain="[('scrap_location', '=', True)]",
+        states={'done': [('readonly', True)]})
```

Comment:
Can you set the scrap location required to True?

Otherwise, it shows the mandatory field validation error while creating a move.

Reply:
@lreficent What about this?

Reply:
I don't see here the comment but I'm pretty sure that @rousseldenis asked for a strict backport of the functionality as it is in v10. That's why this is not readonly, actually I have a custom module to apply that change and some other to this module... 

Anyway odoo has fixed this https://github.com/odoo/odoo/commit/4d86cc78f16271a077b1456a3adcf6b2b3372390 some time after my PR so I think I can add it now also here.